### PR TITLE
Remove .travis.yml

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Travis is disabled for a while now and there are no plans to activate it
again.

### CODE REVIEW

**Comment by sschuberth** [comment]
> Travis is disabled for a while now and there are no plans to activate it
> again.

Could you please add to the commit message that the reason for not having plans to reactivate is their change in terms of use as announced at https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing#building-on-a-public-repositories-only?

Also, please rebase to make the Docker build error go away.

**Comment by sschuberth** [comment]
Also, note that Travis was the only CI that was uploading to CodeCov for the coverage report. I actually liked the coverage badge. So before we delete Travis completely, we should migrate the CodeCov upload to an Azure step, I believe,

**Comment by mnonnenmacher** [comment]
@sschuberth I have added a commit that moves the codecov config to Azure Pipelines.

File: .azure-pipelines/LinuxTest.yml [hunk 1]
```diff
@@ -105,3 +105,7 @@ jobs:
   # Ensure that any running Gradle daemon is stopped before caching starts.
   - bash: ./gradlew --stop
     displayName: Stop Gradle Daemon
+
+  - script: bash <(curl -s https://codecov.io/bash) -X gcov -f '!*/src/*/assets/*';
+    displayName: Upload to codecov.io
+    condition: eq(variables['Build.Reason'], 'Schedule')
```
Comment (sschuberth) [inline .azure-pipelines/LinuxTest.yml:111]: Hmm, so the upload is now split into two places. Will the statistics then be merged on the server side? If not, which of the two does the percentage in the coverage badge then refer to?

Comment (mnonnenmacher) [reply .azure-pipelines/LinuxTest.yml:111]: Codecov does automatically merge multiple uploads: https://docs.codecov.io/docs/merging-reports

Comment (sschuberth) [reply .azure-pipelines/LinuxTest.yml:111]: Nice. Could you add that info to the commit message please?

Comment (mnonnenmacher) [reply .azure-pipelines/LinuxTest.yml:111]: Done.

**Review by sschuberth — APPROVED** [review]

**Comment by sschuberth** [comment]
BTW, should we remove `.appveyor.yml` now, too? The AppVeyor build has been very flaky recently anyway.

**Comment by mnonnenmacher** [comment]
> BTW, should we remove `.appveyor.yml` now, too? The AppVeyor build has been very flaky recently anyway.

I'm fine with removing it, but it would mean you have to get used to the Azure test reports ;)

**Comment by sschuberth** [comment]
> Azure test reports

These still suck indeed. But I'm still fine with getting rid of `.appyveor.yml` for the sake of combining all our CI checks in Azure. Can you do the removal?

**Comment by sschuberth** [comment]
I've done the removal in https://github.com/oss-review-toolkit/ort/pull/3611.

**Pull request merged**